builtin/mainmenu/tab_local.lua

```diff
@@ -172,8 +172,8 @@ local function get_formspec(tabview, name, tabdata)
 	local world = list and index and list[index]
 	local game
 
-	local is_world_selected = list and list[core.get_textlist_index("sp_worlds") or index]
-
+	local list_selected = core.get_textlist_index("sp_worlds") or index or 0
+	local is_world_selected = list and (list[list_selected] or list_selected > #list)
```

Comment:
In which case would `list_selected > #list` be a valid selection?

Reply:
I see, so the formspec fields are yet not sent with the new formspec, resulting in a mismatch between A) what's shown in the GUI (limited to max count) and B) the cached selection position.